pgtype/json_test.go

```diff
@@ -224,3 +225,20 @@ func TestJSONCodecEncodeJSONMarshalerThatCanBeWrapped(t *testing.T) {
 		require.Equal(t, `{"custom":"thing"}`, jsonStr)
 	})
 }
+
+func TestJSONCodecCustomMarshal(t *testing.T) {
+	connTestRunner := defaultConnTestRunner
+	connTestRunner.AfterConnect = func(ctx context.Context, t testing.TB, conn *pgx.Conn) {
+		conn.TypeMap().RegisterType(&pgtype.Type{
+			Name: "json", OID: pgtype.JSONOID, Codec: &pgtype.JSONCodec{
+				Marshal: func(v any) ([]byte, error) {
+					return []byte(`{"custom":"value"}`), nil
+				},
+				Unmarshal: json.Unmarshal,
+			}})
+	}
+
+	pgxtest.RunValueRoundTripTests(context.Background(), t, connTestRunner, pgxtest.KnownOIDQueryExecModes, "json", []pgxtest.ValueRoundTripTest{
+		{[]byte(`"hello"`), new(string), isExpectedEq(`{"custom":"value"}`)},
```

Comment:
@jackc I do not know how to write this test. This does not work. It seems like Marshal is simply not called? Any insight? Maybe I should not use `RunValueRoundTripTests`?